### PR TITLE
📖 Update description of transport controller in architecture doc

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR updates the architecture document, regarding the transport controller. This PR improves accuracy and expands on some details.

## Related issue(s)

Fixes #

### CODE REVIEW

File: docs/content/direct/architecture.md [hunk 1]
```diff
@@ -102,11 +102,11 @@ in the [KubeFlex Architecture](https://github.com/kubestellar/kubeflex/blob/main
 There are currently two roles for spaces managed by KubeFlex: Inventory and Transport Space 
 (ITS) and Workload Description Space (WDS). The former runs the [OCM Cluster Manager](#ocm-cluster-manager) on a vcluster-type control plane, and the latter runs on a k8s-type control plane.
 
-An ITS holds the OCM inventory (`ManagedCluster`) objects and mailbox namespaces. The mailbox namespaces and their contents are implementation details that users do not deal with. Each mailbox namespace corresponds 1:1 with a WEC and holds `ManifestWork` objects managed by the central KubeStellar controllers.
+An ITS holds the inventory and the mailbox namespaces. The inventory is anchored by `ManagedCluster` objects that describe the WECs. For each WEC there may also be a `ConfigMap` object (in the `customization-properties` namespace) that carries additional properties of that WEC; this `ConfigMap` is used in cutomizing the workload to the WEC. The mailbox namespaces and their contents are transport implementation details that users do not deal with. Each mailbox namespace corresponds 1:1 with a WEC and holds `ManifestWork` objects managed by the central KubeStellar controllers.
```
Comment (pdettori) [inline docs/content/direct/architecture.md:106]: s/cutomizing/customizing/

Comment (pdettori) [reply docs/content/direct/architecture.md:106]: s/ users do not deal with/ users do not need to deal with/

File: docs/content/direct/architecture.md [hunk 2]
```diff
@@ -102,11 +102,11 @@ in the [KubeFlex Architecture](https://github.com/kubestellar/kubeflex/blob/main
...
 
 A WDS holds user workload objects and the user's objects that form the interface to KubeStellar control. 
 Currently, the user control objects are `BindingPolicy` and `Binding` objects.
-We plan to later add objects to specify customization and summarization.
+Future development may define more kinds of control objects that exist in the WDS.
```
Comment (pdettori) [inline docs/content/direct/architecture.md:110]: Perhaps better to say "hosted in the WDS" instead than "that exist in the WDS"

File: docs/content/direct/architecture.md [hunk 3]
```diff
@@ -466,5 +466,16 @@ The following section describes how transport controller works, while the descri
...
+
+The transport controller watches for `Binding` objects on the WDS, using an informer. Upon every add, update, and delete event from that informer, the controller puts a reference to that `Binding` object in its work queue. The transport controller also has informers on the inventory objects (both `ManagedCluster` and their associated `ConfigMap`) and on the wrapped objects (`ManifestWork`). Forked goroutines process items from the work queue. For a reference to a control or workload object, that processing starts with retrieving the informer's cached copy of that object. 
+
+The transport controller also maintains a finalizer on each Binding object. When processing a reference to a `Binding` object that no longer exists, the transport controller has nothing more to do --- the work was done earlier before removing that finalizer.
+
```
Comment (pdettori) [inline docs/content/direct/architecture.md:474]: s/---/;/

File: docs/content/direct/architecture.md [hunk 4]
```diff
@@ -466,5 +466,16 @@ The following section describes how transport controller works, while the descri
...
...ontroller reads those objects from the WDS using a Kubernetes "dynamic" client. The controller goes through those objects one-by-one, applying template expansion for each destination if the object requests template expansion. If any of those objects requests template expansion and has a leaf string that actually involves template expansion: the controller accumulates a map from destination to slice of customized objects and then invokes the transport plugin on each of those slices, to ultimately produce a function from destination to wrapped object. If none of the selected workload objects actually involved any template expansion then the controller wraps the slice of workload objects to get one wrapped object and produces a constant function from destination to that one wrapped object. The controller then iterates over the `Binding`'s list of desinations and propagates the corresponding wrapped object (reported by the function just described) to the corresponding mailbox namespace.  Once the wrapped object is in the mailbox namespace of a cluster on the ITS, it's the agent responsibility to pull the wrapped object from there and apply/update/delete the workload objects on the WEC.
```
Comment (vMaroon) [inline docs/content/direct/architecture.md:477]: 1. On `leaf string`: having a leaf string implies having non-leaf strings. Both are not clear from this file alone.
2. This paragraph requires caching much information on the go, perhaps introducing bullets to better separate logic can help.

File: docs/content/direct/architecture.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/direct/architecture.md:106]: done

File: docs/content/direct/architecture.md (hunk 2 above)
Comment (MikeSpreitzer) [reply docs/content/direct/architecture.md:110]: done

File: docs/content/direct/architecture.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/direct/architecture.md:474]: reworded, a little differently.

File: docs/content/direct/architecture.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/direct/architecture.md]: I took out the word "leaf".

I factored that big paragraph.

**Comment by vMaroon** [comment]
Looks good to me, @pdettori up to you now.

**Pull request merged**